### DISCUSSION TITLE
How can i change this code use lightning?

### DISCUSSION THREAD

**Discussion by morestart** [Lightning-AI/pytorch-lightning#10007] [General]
```Python
def evaluate(net, val_loader, loss_func, to_use_device, logger, converter, metric):
    logger.info('start evaluate')
    net.eval()
    nums = 0
    result_dict = {'eval_loss': 0., 'eval_acc': 0., 'norm_edit_dis': 0.}
    show_str = []
    with torch.no_grad():
        for batch_data in tqdm(val_loader):
            targets, targets_lengths = converter.encode(batch_data['label'])
            batch_data['targets'] = targets
            batch_data['targets_lengths'] = targets_lengths
            output = net.forward(batch_data['img'].to(to_use_device))
            loss = loss_func(output, batch_data)

            nums += batch_data['img'].shape[0]
            acc_dict = metric(output, batch_data['label'])
            result_dict['eval_loss'] += loss['loss'].item()
            result_dict['eval_acc'] += acc_dict['n_correct']
            result_dict['norm_edit_dis'] += acc_dict['norm_edit_dis']
            show_str.extend(acc_dict['show_str'])

    result_dict['eval_loss'] /= len(val_loader)
    result_dict['eval_acc'] /= nums
    result_dict['norm_edit_dis'] = 1 - result_dict['norm_edit_dis'] / nums
    logger.info(f"eval_loss:{result_dict['eval_loss']}")
    logger.info(f"eval_acc:{result_dict['eval_acc']}")
    logger.info(f"norm_edit_dis:{result_dict['norm_edit_dis']}")

    for s in show_str[:10]:
        logger.info(s)
    net.train()
    return result_dict
```

if use `validation_step` func, i can't get `len(val_loader)` num... and like `nums` var maybe i need to use  globle var to save?

**Comment by rohitgr7** [comment]
you can just do `self.log('eval_loss', loss['loss'])` inside `validation_step` and it will automatically accumulate this for you. You can access this accumulated value using `self.trainer.callback_metrics` inside `validation_epoch_end`.
